src/nbl/asset/interchange/IAssetLoader.cpp

```diff
@@ -35,22 +35,36 @@ void IAssetLoader::IAssetLoaderOverride::insertAssetIntoCache(SAssetBundle& asse
         m_manager->insertAssetIntoCache(asset, ASSET_MUTABILITY_ON_CACHE_INSERT);
 }
 
-bool IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
+void IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
```

Comment:
you've rvalue moved the chosen asset into the function, the function should be able to return a different chosen asset (smart pointer) 

but the default implementation should obvs just return `_chosenAsset` on success, and `nullptr` on failure

Reply:
on a second thought, does anyone else except for `getAssetInHierarchy_impl` call this function?

Reply:
> on a second thought, does anyone else except for getAssetInHierarchy_impl call this function?

nah